core/ivr/ivr.go

```diff
@@ -411,12 +411,12 @@ func ResumeIVRFlow(
 		return fmt.Errorf("error creating flow contact: %w", err)
 	}
 
-	session, err := models.FindWaitingSessionForContact(ctx, rt, oa, models.FlowTypeVoice, contact)
+	session, err := models.FindWaitingSessionForContact(ctx, rt, oa, c, contact)
 	if err != nil {
 		return fmt.Errorf("error loading session for contact: %w", err)
 	}
 
-	if session == nil {
+	if session == nil || session.SessionType() != models.FlowTypeVoice {
```

Comment:
this is resuming a voice flow - it's an error if the waiting session is a message flow